apps/prairielearn/assets/scripts/navbarClient.ts

```diff
@@ -0,0 +1,154 @@
+    Cookies.remove('pl_requested_course_instance_role', { path: '/' });
+    Cookies.remove('pl_requested_mode', { path: '/' });
+    Cookies.remove('pl_requested_date', { path: '/' });
+    Cookies.set('pl_requested_data_changed', 'true', { path: '/' });
+
+    if (viewType === 'instructor') {
+      location.reload();
+    }
+  });
+
+  document.querySelector('#navbar-user-view-authn-instructor')?.addEventListener('click', () => {
+    Cookies.remove('pl_requested_uid', { path: '/' });
+    Cookies.remove('pl_requested_course_role', { path: '/' });
+    Cookies.remove('pl_requested_course_instance_role', { path: '/' });
+    Cookies.set('pl_requested_data_changed', 'true', { path: '/' });
+
+    if (viewType === 'instructor') {
+      location.reload();
+    }
+  });
+
+  document
+    .querySelector('#navbar-user-view-authn-student-no-rules')
+    ?.addEventListener('click', () => {
+      Cookies.remove('pl_requested_uid', { path: '/' });
+      Cookies.remove('pl_requested_course_role', { path: '/' });
+      Cookies.remove('pl_requested_course_instance_role', { path: '/' });
+      Cookies.set('pl_requested_data_changed', 'true', { path: '/' });
+
+      if (viewType === 'student') {
+        location.reload();
+      }
+    });
+
+  document.querySelector('#navbar-user-view-authn-student')?.addEventListener('click', () => {
+    Cookies.remove('pl_requested_uid', { path: '/' });
+    Cookies.set('pl_requested_course_role', 'None', { path: '/', expires: expires });
+    Cookies.set('pl_requested_course_instance_role', 'None', { path: '/', expires: expires });
+    Cookies.set('pl_requested_data_changed', 'true', { path: '/' });
+
+    if (viewType === 'student') {
+      location.reload();
+    }
+  });
+
+  document.querySelector('#navbar-user-view-instructor')?.addEventListener('click', () => {
+    Cookies.remove('pl_requested_course_role', { path: '/' });
+    Cookies.remove('pl_requested_course_instance_role', { path: '/' });
+    Cookies.set('pl_requested_data_changed', 'true', { path: '/' });
+
+    if (viewType === 'instructor') {
+      location.reload();
+    }
+  });
+
+  document.querySelector('#navbar-user-view-student-no-rules')?.addEventListener('click', () => {
+    Cookies.remove('pl_requested_course_role', { path: '/' });
+    Cookies.remove('pl_requested_course_instance_role', { path: '/' });
+    Cookies.set('pl_requested_data_changed', 'true', { path: '/' });
+
+    if (viewType === 'student') {
+      location.reload();
+    }
+  });
+
+  document.querySelector('#navbar-user-view-student')?.addEventListener('click', () => {
+    const courseRole = hasInstructorAccess ? 'None' : authnCourseRole;
+    const courseInstanceRole = hasInstructorAccess ? 'None' : authnCourseInstanceRole;
+
+    Cookies.set('pl_requested_course_role', courseRole, { path: '/', expires: expires });
+    Cookies.set('pl_requested_course_instance_role', courseInstanceRole, {
+      path: '/',
+      expires: expires,
+    });
```

Comment:
This is mixing up the behaviour of `#navbar-user-view-student` and `#navbar-user-view-student-no-rules`. The correct behaviour is that, if the UID has instructor access, then `...-student-no-rules` should remove the cookie and `...-student` should set the cookie to None, while if the UID does not have that, then `...-student-no-rules` should set the cookie and `...-student` should remove it. In practice the code above does not work if the effective UID is a student and I want to run "without user restrictions", since it sets the cookie to None instead of the authenticated user's role.

Reply:
Thank you for looking at this so carefully. I believe I fixed this behavior in 4c0df92, could you check to make sure it looks how you expect it to now?

Reply:
Yup, all good now.